File: cash/cash.c
// Calculates the minimum number of coins with which a change can be made

#include <cs50.h>
#include <stdio.h>
#include <math.h>

float get_input(void);
int get_coins(float);

int main(void)
{
    float change_owed = get_input();
    printf("%i\n", get_coins(change_owed));
}

// Prompts the user for a non-negative change
float get_input(void)
{
    float change_owed;
    do
    {
        change_owed = get_float("Change owed: ");
    }
    while (change_owed < 0);
    return change_owed;
}

// Returns number of coins
int get_coins(float change)
{
    // Converts dollars into cents
    int cents = round(change * 100);
    
    // Calculates number of quarters
    int coins = cents / 25;
    cents %= 25;
    
    // Calculates number of dimes
    coins += cents / 10;
    cents %= 10;
    
    // Calculates number of nickels
    coins += cents / 5;
    cents %= 5;
    
    // Calculates number of pennies
    coins += cents;
    
    return coins;
}

File: cash/cash.py
from cs50 import get_float


def get_positive_change_owed():
    # Function to prompt user for non-negative change owed

    while True:
        change_owed = get_float("Change owed: ")
        if change_owed > 0:
            return change_owed


def get_coins(change_owed):
    # Function to calculate minimum number of coins that can be made

    # Convert dollars into cents
    cents = round(change_owed * 100)

    # Calculate number of quarters
    coins = cents // 25
    cents %= 25

    # Calculate number of dimes
    coins += cents // 10
    cents %= 10

    # Calculate number of nickels
    coins += cents // 5
    cents %= 5

    # Calculate number of pennies
    coins += cents

    return coins


def main():
    change_owed = get_positive_change_owed()
    print(get_coins(change_owed))


if __name__ == "__main__":
    # Call main function
    main()